test/state/state.cpp

```diff
@@ -460,6 +480,85 @@ TransactionReceipt transition(const StateView& state_view, const BlockInfo& bloc
     assert(sender_acc.nonce < Account::NonceMax);  // Required for valid tx.
     ++sender_acc.nonce;                            // Bump sender nonce.
 
+    auto delegation_refund = 0;
+
+    for (const auto& auth : tx.authorization_list)
+    {
+        if (auth.chain_id != 0 && auth.chain_id != tx.chain_id)
+            continue;
+
+        if (auth.nonce == Account::NonceMax)
+            continue;
+
+        // Check if signer could be ecrecovered.
+        if (!auth.signer.has_value())
+            continue;
+
+        if (auth.s > SECP256K1N_OVER_2)
+            continue;
+
+        // Check if authority exists.
+        auto* authority_ptr = state.find(*auth.signer);
+        if (authority_ptr != nullptr && !authority_ptr->is_empty())
+        {
+            authority_ptr->access_status = EVMC_ACCESS_WARM;
+
+            // Skip if authority has non-delegated code.
+            if (authority_ptr->code_hash != Account::EMPTY_CODE_HASH &&
+                !is_code_delegated(state.get_code(*auth.signer)))
+                continue;
+
+            // Skip if authorization nonce is incorrect.
+            if (auth.nonce != authority_ptr->nonce)
+                continue;
+        }
+        else
+        {
+            // Create authority account.
+            // It is still considered empty at this point until nonce increment, but already warm.
+            authority_ptr = &state.get_or_insert(
+                *auth.signer, {.erase_if_empty = true, .access_status = EVMC_ACCESS_WARM});
+
+            // Skip if authorization nonce is incorrect.
+            if (auth.nonce != 0)
+                continue;
+        }
+
+        // Refund if authority account creation is not needed,
+        // i.e. if it had non-empty balance, nonce, code or storage.
+        if (authority_ptr != nullptr &&
+            (!authority_ptr->is_empty() || authority_ptr->has_initial_storage))
```

Comment:
Inspecting storage is quite problematic here. There should be a better way to answer the question of the account is in the trie.